### PR TITLE
WIP add spring configuration module & spring boot autoconfiguration

### PR DESCRIPTION
**Pull request by uweschaefer**

### CODE REVIEW

File: cryptoshred-spring/src/main/java/eu/prismacapacity/cryptoshred/spring/CryptoShredConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		CryptoObjectMapper.Builder b = CryptoObjectMapper.builder(repository, engine);
+
+		if (metrics != null)
+			b.metrics(metrics);
+
+		if (om != null)
+			b.mapper(om);
+
+		if (size < 128) {
+			throw new IllegalArgumentException("keySize is out or range: " + size);
```
Comment (otbe) [inline cryptoshred-spring/src/main/java/eu/prismacapacity/cryptoshred/spring/CryptoShredConfiguration.java]: `of` 🤔

File: cryptoshred-spring-boot-autoconfigure/pom.xml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      <artifactId>cryptoshred-core</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>eu.prismacapacity</groupId>
+      <artifactId>cryptoshred-spring</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>eu.prismacapacity</groupId>
```
Comment (otbe) [inline cryptoshred-spring-boot-autoconfigure/pom.xml]: Is this really necessary? 🤔

File: cryptoshred-spring-cloud-aws/src/main/java/eu/prismacapacity/cryptoshred/spring/cloud/aws/AWSDynamoDBKeyRepositoryConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+import eu.prismacapacity.cryptoshred.cloud.aws.DynamoDBCryptoKeyRepository;
+import eu.prismacapacity.cryptoshred.core.CryptoEngine;
+import eu.prismacapacity.cryptoshred.core.keys.CryptoKeyRepository;
+
+@Configuration
+public class AWSDynamoDBKeyRepositoryConfiguration {
+	@Bean
+	public CryptoKeyRepository cryptoKeyRepository(CryptoEngine engine, AmazonDynamoDB dynamoDB,
+
```
Comment (otbe) [inline cryptoshred-spring-cloud-aws/src/main/java/eu/prismacapacity/cryptoshred/spring/cloud/aws/AWSDynamoDBKeyRepositoryConfiguration.java]: Why is here a blank line? But nevertheless we finally have spotless in place 😂👍

File: cryptoshred-spring/src/main/java/eu/prismacapacity/cryptoshred/spring/CryptoShredConfiguration.java (hunk 1 above)
Comment (uweschaefer) [reply cryptoshred-spring/src/main/java/eu/prismacapacity/cryptoshred/spring/CryptoShredConfiguration.java]: :D

File: cryptoshred-spring-cloud-aws/src/main/java/eu/prismacapacity/cryptoshred/spring/cloud/aws/AWSDynamoDBKeyRepositoryConfiguration.java (hunk 3 above)
Comment (uweschaefer) [reply cryptoshred-spring-cloud-aws/src/main/java/eu/prismacapacity/cryptoshred/spring/cloud/aws/AWSDynamoDBKeyRepositoryConfiguration.java]: oh come on...

File: cryptoshred-spring-boot-autoconfigure/pom.xml (hunk 2 above)
Comment (uweschaefer) [reply cryptoshred-spring-boot-autoconfigure/pom.xml]: yes, i think so. please elaborate. Note that transitive does not work for optional deps.

Comment (otbe) [reply cryptoshred-spring-boot-autoconfigure/pom.xml]: Mh yes, but on the client site you have to define a dependency to `cryptoshred-spring-cloud-aws` anyways which includes `cryptoshred-cloud-aws` so we should be fine without it, shouldnt we?

If not I guess we need a dependency to `cryptoshred-micrometer` as well?! 🤔

Comment (uweschaefer) [reply cryptoshred-spring-boot-autoconfigure/pom.xml]: good point.

**Pull request merged**